enterprise/tasks.py

```diff
@@ -49,8 +49,16 @@ def create_enterprise_enrollment(course_id, enterprise_customer_user_id):
         # We have made changes elsewhere to avoid this issue, but in the mean time
         # we believe a Source of ENROLLMENT_TASK is more clear.
 
-        EnterpriseCourseEnrollment.objects.create(
-            course_id=course_id,
-            enterprise_customer_user=enterprise_customer_user,
-            source=EnterpriseEnrollmentSource.get_source(EnterpriseEnrollmentSource.ENROLLMENT_TASK)
-        )
+        try:
+            EnterpriseCourseEnrollment.objects.get_or_create(
```

Comment:
Making this bit more defensible given the

https://github.com/edx/edx-enterprise/pull/1165/files#diff-d950991245f878f58adc157895d6431af4385d522ab994aa25726e9abc2b434dR1037-R1045